# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: image/decoders/nsAVIFDecoder.h
# Commit: da557d1806dd
# Full Hash: da557d1806dd9e1e519e00a4481b2e075f722ae8
# Author: Butkovits Atila <abutkovits@mozilla.com>
# Date: 2023-01-27 22:54 +0200
# Regressor Bug: 1788119
# File Overlap Count: 1
# Overlapping Files: Cargo.lock
# Description:
#   Backed out 5 changesets (bug 1788119) for causing bustages complaining about AVIFDecodedData. CLOSED TREE
# ==============================================================================

diff -r d980bebc6d26 -r da557d1806dd image/decoders/nsAVIFDecoder.h
--- a/image/decoders/nsAVIFDecoder.h	Fri Jan 27 20:35:34 2023 +0000
+++ b/image/decoders/nsAVIFDecoder.h	Fri Jan 27 22:54:08 2023 +0200
@@ -8,10 +8,7 @@
 #define mozilla_image_decoders_nsAVIFDecoder_h
 
 #include "Decoder.h"
-#include "mozilla/gfx/Types.h"
-#include "MP4Metadata.h"
 #include "mp4parse.h"
-#include "SampleIterator.h"
 #include "SurfacePipe.h"
 
 #include "aom/aom_decoder.h"
@@ -22,9 +19,6 @@
 namespace mozilla {
 namespace image {
 class RasterImage;
-class AVIFDecoderStream;
-class AVIFParser;
-class AVIFDecoderInterface;
 
 class nsAVIFDecoder final : public Decoder {
  public:
@@ -40,7 +34,6 @@
  private:
   friend class DecoderFactory;
   friend class AVIFDecoderInterface;
-  friend class AVIFParser;
 
   // Decoders should only be instantiated via DecoderFactory.
   explicit nsAVIFDecoder(RasterImage* aImage);
@@ -53,8 +46,8 @@
   typedef Variant<aom_codec_err_t, NonAOMCodecError> AOMResult;
   enum class NonDecoderResult {
     NeedMoreData,
-    OutputAvailable,
-    Complete,
+    MetadataOk,
+    NoPrimaryItem,
     SizeOverflow,
     OutOfMemory,
     PipeInitError,
@@ -63,12 +56,9 @@
     AlphaYColorDepthMismatch,
     MetadataImageSizeMismatch,
     InvalidCICP,
-    NoSamples,
   };
   using DecodeResult =
       Variant<Mp4parseStatus, NonDecoderResult, Dav1dResult, AOMResult>;
-  Mp4parseStatus CreateParser();
-  DecodeResult CreateDecoder();
   DecodeResult Decode(SourceBufferIterator& aIterator, IResumable* aOnResume);
 
   static bool IsDecodeSuccess(const DecodeResult& aResult);
@@ -76,106 +66,9 @@
   void RecordDecodeResultTelemetry(const DecodeResult& aResult);
 
   Vector<uint8_t> mBufferedData;
-  UniquePtr<AVIFDecoderStream> mBufferStream;
 
   /// Pointer to the next place to read from mBufferedData
   const uint8_t* mReadCursor = nullptr;
-
-  UniquePtr<AVIFParser> mParser = nullptr;
-  UniquePtr<AVIFDecoderInterface> mDecoder = nullptr;
-
-  bool mIsAnimated = false;
-  bool mHasAlpha = false;
-};
-
-class AVIFDecoderStream : public ByteStream {
- public:
-  explicit AVIFDecoderStream(Vector<uint8_t>* aBuffer) { mBuffer = aBuffer; }
-
-  virtual bool ReadAt(int64_t offset, void* data, size_t size,
-                      size_t* bytes_read) override;
-  virtual bool CachedReadAt(int64_t offset, void* data, size_t size,
-                            size_t* bytes_read) override {
-    return ReadAt(offset, data, size, bytes_read);
-  };
-  virtual bool Length(int64_t* size) override;
-  virtual const uint8_t* GetContiguousAccess(int64_t aOffset,
-                                             size_t aSize) override;
-
- private:
-  Vector<uint8_t>* mBuffer;
-};
-
-struct AVIFImage {
-  uint32_t mFrameNum = 0;
-  FrameTimeout mDuration = FrameTimeout::Zero();
-  RefPtr<MediaRawData> mColorImage = nullptr;
-  RefPtr<MediaRawData> mAlphaImage = nullptr;
-};
-
-class AVIFParser {
- public:
-  static Mp4parseStatus Create(const Mp4parseIo* aIo, ByteStream* aBuffer,
-                               UniquePtr<AVIFParser>& aParserOut);
-
-  ~AVIFParser();
-
-  const Mp4parseAvifInfo& GetInfo() const { return mInfo; }
-
-  nsAVIFDecoder::DecodeResult GetImage(AVIFImage& aImage);
-
-  bool IsAnimated() const;
-
- private:
-  explicit AVIFParser(const Mp4parseIo* aIo);
-
-  Mp4parseStatus Init(ByteStream* aBuffer);
-
-  struct FreeAvifParser {
-    void operator()(Mp4parseAvifParser* aPtr) { mp4parse_avif_free(aPtr); }
-  };
-
-  const Mp4parseIo* mIo;
-  UniquePtr<Mp4parseAvifParser, FreeAvifParser> mParser = nullptr;
-  Mp4parseAvifInfo mInfo = {};
-
-  UniquePtr<SampleIterator> mColorSampleIter = nullptr;
-  UniquePtr<SampleIterator> mAlphaSampleIter = nullptr;
-  uint32_t mFrameNum = 0;
-};
-
-struct AVIFDecodedData;
-
-// An interface to do decode and get the decoded data
-class AVIFDecoderInterface {
- public:
-  using Dav1dResult = nsAVIFDecoder::Dav1dResult;
-  using NonAOMCodecError = nsAVIFDecoder::NonAOMCodecError;
-  using AOMResult = nsAVIFDecoder::AOMResult;
-  using NonDecoderResult = nsAVIFDecoder::NonDecoderResult;
-  using DecodeResult = nsAVIFDecoder::DecodeResult;
-
-  virtual ~AVIFDecoderInterface();
-
-  // Set the mDecodedData if Decode() succeeds
-  virtual DecodeResult Decode(bool aIsMetadataDecode,
-                              const Mp4parseAvifInfo& aAVIFInfo,
-                              const AVIFImage& aSamples) = 0;
-  // Must be called only once after Decode() succeeds
-  UniquePtr<AVIFDecodedData> GetDecodedData() {
-    MOZ_ASSERT(mDecodedData);
-    return std::move(mDecodedData);
-  }
-
- protected:
-  explicit AVIFDecoderInterface() = default;
-
-  inline static bool IsDecodeSuccess(const DecodeResult& aResult) {
-    return nsAVIFDecoder::IsDecodeSuccess(aResult);
-  }
-
-  // The mDecodedData is valid after Decode() succeeds
-  UniquePtr<AVIFDecodedData> mDecodedData;
 };
 
 }  // namespace image